cuebot/build.gradle

```diff
@@ -40,24 +43,23 @@ dependencies {
     compile group: 'org.apache.activemq', name: 'activemq-pool', version: activemqVersion
     compile group: 'org.apache.velocity', name: 'velocity', version: '1.7'
     compile group: 'org.jdom', name: 'jdom', version: '1.1.3'
-    compile group: 'org.springframework.boot', name: 'spring-boot-starter-jdbc', version: springBootVersion
-    compile group: 'org.springframework.boot', name: 'spring-boot-starter-log4j', version: springBootVersion
-    compile group: 'org.springframework.boot', name: 'spring-boot-starter-web', version: springBootVersion
-    compile group: 'org.springframework', name: 'spring-context-support', version: springVersion
-    compile group: 'org.springframework', name: 'spring-jms', version: springVersion
+    compile group: 'org.springframework.boot', name: 'spring-boot-starter-jdbc'
+    compile group: 'org.springframework.boot', name: 'spring-boot-starter-web'
+    compile group: 'org.springframework', name: 'spring-context-support'
+    compile group: 'org.springframework', name: 'spring-jms'
```

Comment:
IIUC all dependencies here should be locked to a specific version.

Reply:
In spring that is not recommended, they have this thing called "opinionated dependencies" or something like that.   Basically unless you keep it in sync manually (which nobody does or does wrong), then you end up with multiple versions. The spring boot dependencies plugins will automatically bring in the correct versions.  In maven, you do the same thing with the <dependencyManagement> section, it basically allows you to define downstream dependencies.

Reply:
> The spring boot dependencies plugins will automatically bring in the correct versions.

Could you explain a bit what you mean by this? Does it automatically use `springBootVersion` defined above somehow?

Reply:
This plugin looks at the version of spring boot we brought in at the top of the file and brings in the version of the dependencies the spring developers tested with.  This extends to even non-spring dependencies as well.

apply plugin: 'io.spring.dependency-management'
https://plugins.gradle.org/plugin/io.spring.dependency-management